### PR TITLE
Prevent warnings about unneeded copy in for loops

### PR DESCRIPTION
**Pull request by fgaray**
Just two small changes to prevent copy of std::pair in for loops

### CODE REVIEW

File: src/cts/src/Clock.cpp [hunk 1]
```diff
@@ -95,7 +95,7 @@ Box<double> Clock::computeSinkRegionClustered(
   auto yMin = std::numeric_limits<float>::max();
   auto yMax = std::numeric_limits<float>::lowest();
 
-  for (const auto [sinkX, sinkY] : sinks) {
+  for (const auto& [sinkX, sinkY] : sinks) {
```
Comment (maliberty) [inline src/cts/src/Clock.cpp:98]: These are just float so there is no copying

File: src/mpl2/src/hier_rtlmp.cpp [hunk 2]
```diff
@@ -3034,7 +3034,7 @@ void HierRTLMP::multiLevelMacroPlacement(Cluster* parent)
   }
   // add the virtual connections (the weight related to IOs and macros belong to
   // the same cluster)
-  for (const auto [cluster1, cluster2] : parent->getVirtualConnections()) {
+  for (const auto& [cluster1, cluster2] : parent->getVirtualConnections()) {
```
Comment (maliberty) [inline src/mpl2/src/hier_rtlmp.cpp:3037]: These are just ints so there is no copying here either.

File: src/cts/src/Clock.cpp (hunk 1 above)
Comment (gadfort) [reply src/cts/src/Clock.cpp:98]: this does generate a warning though on my builds

Comment (fgaray) [reply src/cts/src/Clock.cpp:98]: Sinks is a vector of std::pair, the unpack of the floats inside the vector is doing a copy of the std::pair before assignment to the floats, that's why we need the reference. It is a reference to the std::pair, not the floats.

Comment (fgaray) [reply src/cts/src/Clock.cpp:98]: My gcc is "gcc (Debian 12.2.0-10) 12.2.0"

**Comment by maliberty** [comment]
I looked in godbolt and the assembly is the same either way and the warning goes away so I'll go for it.  It seems a dumb warning for primitive types.

**Review by maliberty — APPROVED** [review]

**Pull request merged**